Source/WebCore/dom/AbortSignal.cpp

```diff
@@ -126,25 +128,50 @@ void AbortSignal::signalAbort(JSC::JSValue reason)
     // 1. If signal's aborted flag is set, then return.
     if (m_aborted)
         return;
-    
+
     // 2. Set signal’s aborted flag.
+    markAborted(reason);
+
+    Vector<Ref<AbortSignal>> dependentSignalsToAbort = { };
+
+    for (Ref dependentSignal : std::exchange(m_dependentSignals, { })) {
+        if (!dependentSignal->aborted()) {
+            dependentSignal->markAborted(reason);
+            dependentSignalsToAbort.append(dependentSignal);
```

Comment:
`WTFMove(dependentSignal)`

Reply:
Thank you. 🙏 I have pushed the nit suggestions. 

Though for my own education, why did we move here? Is it because the m_dependentSignals is emptied and thus ref is --, so we need something else to hold it?

Reply:
This is a performance optimization, it avoids ref-counting churn. We've moving `dependentSignal` straight into the vector, without increasing the ref-count. We can do do since `dependentSignal` is not used after appending to the vector.

Without moving, appending to the vector would increase `dependentSignal`'s refcount, then the refcount would be decreased when the variables goes out of scope (next loop iteration).